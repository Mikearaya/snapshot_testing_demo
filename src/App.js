import './App.css';

import EventItem from './components/EventItem';
import React from 'react';

const events = [
  {
    id: 1,
    name: 'First Event',
    description:
      'To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting',
    image: 'image,',
    startDate: new Date().toDateString()
  },
  {
    id: 2,
    name: 'Second Event',
    description:
      'To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting',
    image: 'image,',
    startDate: new Date().toDateString()
  },
  {
    id: 3,
    name: 'Second Event',
    description:
      'To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting',
    image: 'image,',
    startDate: new Date().toDateString()
  },
  {
    id: 4,
    name: 'fourth Event',
    description:
      'To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting',
    image: 'image,',
    startDate: new Date().toDateString()
  }
];

function App() {
  return (
    <ul>
      {events.map(e => {
        return (
          <li className='event-items' key={e.id}>
            <EventItem
              name={e.name}
              description={e.description}
              image={e.image}
              startDate={e.startDate}
            />
          </li>
        );
      })}
    </ul>
  );
}

export default App;
