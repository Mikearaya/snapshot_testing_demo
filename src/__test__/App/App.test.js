import App from '../../App';
import React from 'react';
import renderer from 'react-test-renderer';

test('should render correctly', () => {
  const component = renderer.create(<App />).toJSON();
  expect(component).toMatchSnapshot();
});
