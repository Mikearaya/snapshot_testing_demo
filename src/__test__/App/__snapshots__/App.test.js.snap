// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<ul>
  <li
    className="event-items"
  >
    <div
      className="container"
    >
      <p>
        First Event
      </p>
      <span
        className="start-date"
      >
         
        Tue Mar 03 2020
      </span>
      <p>
        To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting
      </p>
    </div>
  </li>
  <li
    className="event-items"
  >
    <div
      className="container"
    >
      <p>
        Second Event
      </p>
      <span
        className="start-date"
      >
         
        Tue Mar 03 2020
      </span>
      <p>
        To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting
      </p>
    </div>
  </li>
  <li
    className="event-items"
  >
    <div
      className="container"
    >
      <p>
        Second Event
      </p>
      <span
        className="start-date"
      >
         
        Tue Mar 03 2020
      </span>
      <p>
        To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting
      </p>
    </div>
  </li>
  <li
    className="event-items"
  >
    <div
      className="container"
    >
      <p>
        fourth Event
      </p>
      <span
        className="start-date"
      >
         
        Tue Mar 03 2020
      </span>
      <p>
        To reduce the number of requests to the server, importing images that are less than 10,000 bytes returns a data URI instead of a path. This applies to the following file extensions: bmp, gif, jpg, jpeg, and png. SVG files are excluded due to #1153. You can control the 10,000 byte threshold by setting
      </p>
    </div>
  </li>
</ul>
`;
